core/modules/flowgen.cc

```diff
@@ -327,13 +344,38 @@ bess::Packet *FlowGen::FillPacket(struct flow *f) {
   if (f->packets_left <= 1)
     tcp_flags |= 0x01; /* FIN */
 
-  *(reinterpret_cast<uint32_t *>(p + 14 + /* IP dst */ 16)) = f->flow_id;
+  if(ip_src_range_ != 0){
```

Comment:
You might instead use the headers in utils/ip.h and utils/tcp.h and then you can access these various fields as a struct rather than manually accessing them. (Unless I'm not understanding what you're trying to do.)